dev-libs/sway/sway-0.6.ebuild

```diff
@@ -0,0 +1,65 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+inherit cmake-utils
+
+DESCRIPTION="i3-compatible Wayland window manager."
+HOMEPAGE="http://swaywm.org/"
+
+SRC_URI="https://github.com/SirCmpwn/sway/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~x86 ~amd64"
+IUSE="X swaylock swaybg swaybar swaygrab swaymsg gdk-pixbuf zsh-completion wallpaper imagemagick ffmpeg systemd"
+
+REQUIRED_USE="swaygrab? ( || ( imagemagick ffmpeg ) )"
+
+CDEPEND="dev-libs/wlc
+		dev-libs/json-c
+		dev-libs/libpcre
+		dev-libs/libinput
+		x11-libs/libxkbcommon
+		dev-libs/wayland
+		x11-libs/pango
+		x11-libs/cairo
+		swaylock? ( virtual/pam )
+		gdk-pixbuf? ( x11-libs/gdk-pixbuf[jpeg] )
+		X? ( dev-libs/wlc[X] )
+		systemd? ( dev-libs/wlc[systemd] sys-apps/systemd )"
+
+RDEPEND="${CDEPEND}
+		X? ( x11-base/xorg-server[wayland] )
+		imagemagick? ( media-gfx/imagemagick[png] )
+		ffmpeg? ( media-video/ffmpeg )"
+
+DEPEND="${CDEPEND}
+		app-text/asciidoc"
+
+src_configure() {
+	local mycmakeargs=(
+		-Denable-swaybar=$(usex swaybar)
+		-Denable-swaybg=$(usex swaybg)
+		-Denable-swaygrab=$(usex swaygrab)
+		-Denable-swaylock=$(usex swaylock)
+		-Denable-swaymsg=$(usex swaymsg)
+
+		-Ddefault-wallpaper=$(usex wallpaper)
+
+		-Denable-gdk-pixbuf=$(usex gdk-pixbuf)
+		-Dzsh-completions=$(usex zsh-completion)
+
+		-DCMAKE_INSTALL_SYSCONFDIR="/etc"
+	)
+
+	cmake-utils_src_configure
+}
+
+src_install() {
+	cmake-utils_src_install
+
+	use !systemd && fperms u+s /usr/bin/sway
```

Comment:
Could Linux capabilities be used instead of setuid?  If so, it'd probably make the more security conscious folks happier.  However, if the equivalent capabilities is the same as setuid, just stick with setuid.